airbyte-integrations/connectors/destination-postgres/src/main/kotlin/io/airbyte/integrations/destination/postgres/client/PostgresAirbyteClient.kt

```diff
@@ -171,14 +201,32 @@ class PostgresAirbyteClient(
     }
 
     override suspend fun discoverSchema(tableName: TableName): TableSchema {
-        TODO("Not yet implemented")
+        val columnsInDb = getColumnsFromDbForDiscovery(tableName)
+        val hasAllAirbyteColumns = columnsInDb.keys.containsAll(COLUMN_NAMES)
+
+        if (!hasAllAirbyteColumns) {
+            val message =
+                "The target table ($tableName) already exists in the destination, but does not contain Airbyte's internal columns. Airbyte can only sync to Airbyte-controlled tables. To fix this error, you must either delete the target table or add a prefix in the connection configuration in order to sync to a separate table in the destination."
+            log.error { message }
+            throw ConfigErrorException(message)
+        }
+
+        // Filter out Airbyte columns
+        val userColumns = columnsInDb.filterKeys { it !in COLUMN_NAMES }
+        return TableSchema(userColumns)
     }
 
     override fun computeSchema(
         stream: DestinationStream,
         columnNameMapping: ColumnNameMapping
     ): TableSchema {
-        TODO("Not yet implemented")
+        val defaultColumnNames = postgresColumnUtils.defaultColumns().map { it.columnName }.toSet()
+        val columns =
+            postgresColumnUtils
+                .getTargetColumns(stream, columnNameMapping)
+                .filter { it.columnName !in defaultColumnNames }
+                .associate { column -> column.columnName to ColumnType(column.columnTypeName, column.nullable) }
+        return TableSchema(columns)
```

Comment:
If you implemented the schema mapper, this should be just be reading the columns off the `stream.tableSchema.columnSchma` like in ClickHouse